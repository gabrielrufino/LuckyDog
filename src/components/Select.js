import React from 'react';
import {Picker} from '@react-native-community/picker';
import {StyleSheet} from 'react-native';
import {View} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    borderColor: '#CACAAA',
    borderRadius: 5,
    borderWidth: 2,
    justifyContent: 'center',
  },
  picker: {
    width: '100%',
  },
});

export default function Select({selected, options, onSelect}) {
  return (
    <View style={styles.container}>
      <Picker
        style={styles.picker}
        selectedValue={selected}
        onValueChange={onSelect}>
        {options.map((option, index) => (
          <Picker.Item key={index} label={option} value={option} />
        ))}
      </Picker>
    </View>
  );
}
