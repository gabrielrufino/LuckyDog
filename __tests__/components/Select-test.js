import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../../src/components/Select';

describe('components/Select.js', () => {
  it('Should renders the select correctly', () => {
    const selected = 'Option 1';
    const options = ['Option 1', 'Option 2', 'Option 3'];
    const onSelect = jest.fn();

    const tree = renderer
      .create(
        <Select selected={selected} options={options} onSelect={onSelect} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
