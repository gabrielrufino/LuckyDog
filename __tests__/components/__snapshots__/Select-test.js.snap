// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select.js Should renders the select correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#CACAAA",
      "borderRadius": 5,
      "borderWidth": 2,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "Selecione um item",
            "textColor": 4288256409,
            "value": undefined,
          },
          Object {
            "label": "Option 1",
            "textColor": undefined,
            "value": "Option 1",
          },
          Object {
            "label": "Option 2",
            "textColor": undefined,
            "value": "Option 2",
          },
          Object {
            "label": "Option 3",
            "textColor": undefined,
            "value": "Option 3",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={1}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
